examples/utils/tf_records_generation.py

```diff
@@ -0,0 +1,189 @@
+flags.DEFINE_string('train_rd_path', '', 'Path to output TFRecord')
+flags.DEFINE_string('test_rd_path', '', 'Path to output TFRecord')
+FLAGS = flags.FLAGS
+
+
+# TO-DO replace this with label map
+def class_text_to_int(row_label):
+    if row_label == 'building':
+        return 1
+    return None
+
+
+def split(df, group):
+    data = namedtuple('data', ['filename', 'object'])
+    gb = df.groupby(group)
+    return [data(filename, gb.get_group(x)) for filename, x in zip(gb.groups.keys(), gb.groups)]
+
+def create_tf_example(group, path):
+    """Creates a tf.Example proto from sample buillding image tile.
+
+Args:
+ encoded_building_image_data: The jpg encoded data of the building image.
+
+Returns:
+ example: The created tf.Example.
+"""
+    with tf.gfile.GFile(op.join(path, '{}'.format(group.filename)), 'rb') as fid:
+        encoded_jpg = fid.read()
+    encoded_jpg_io = io.BytesIO(encoded_jpg)
+    image = Image.open(encoded_jpg_io)
+    width, height = image.size
+    filename = group.filename.encode('utf8')
+    image_format = b'jpg'
+    xmins = []
+    xmaxs = []
+    ymins = []
+    ymaxs = []
+    classes_text = []
+    classes = []
+
+    for _, row in group.object.iterrows():
+        xmins.append(row['xmin'] / width)
+        xmaxs.append(row['xmax'] / width)
+        ymins.append(row['ymin'] / height)
+        ymaxs.append(row['ymax'] / height)
+        classes_text.append(row['class'].encode('utf8'))
+        classes.append(class_text_to_int(row['class']))
+
+    tf_example = tf.train.Example(features=tf.train.Features(feature={
+        'image/height': dataset_util.int64_feature(height),
+        'image/width': dataset_util.int64_feature(width),
+        'image/filename': dataset_util.bytes_feature(filename),
+        'image/source_id': dataset_util.bytes_feature(filename),
+        'image/encoded': dataset_util.bytes_feature(encoded_jpg),
+        'image/format': dataset_util.bytes_feature(image_format),
+        'image/object/bbox/xmin': dataset_util.float_list_feature(xmins),
+        'image/object/bbox/xmax': dataset_util.float_list_feature(xmaxs),
+        'image/object/bbox/ymin': dataset_util.float_list_feature(ymins),
+        'image/object/bbox/ymax': dataset_util.float_list_feature(ymaxs),
+        'image/object/class/text': dataset_util.bytes_list_feature(classes_text),
+        'image/object/class/label': dataset_util.int64_list_feature(classes),
+    }))
+    return tf_example
+
+
+def main(_):
+    labels = np.load(op.join(os.getcwd(), FLAGS.label_input))
+    tile_names = [tile for tile in labels.files]
+    tile_names.sort()
+    tiles = np.array(tile_names)
+
+    tf_tiles_info = []
+
+    for tile in tiles:
+        bboxes = labels[tile].tolist()
+        width = 256
+        height = 256
+        if len(bboxes) != 0:
+            for bbox in bboxes:
+                if bbox[4] == 1:
+                    cl_str = "building"
+                    if bbox[0] < 0:
```

Comment:
@Geoyi can you abstract this part out into a "clamping function" like I showed earlier?

Reply:
updated, thanks for helping out, @drewbo